src/models/brackets.rs

```diff
@@ -371,39 +371,84 @@ impl Bracket {
         }
         let rounds = self.rounds(conn)?;
         let mut races = vec![];
+        let is_rr = match self.bracket_type()? {
+            BracketType::Swiss => false,
+            BracketType::RoundRobin => true,
+        };
+
+        struct StandingsRace {
+            player_1_id: i32,
+            player_2_id: i32,
+            player_1_result: PlayerResult,
+            player_2_result: PlayerResult,
+            outcome: Outcome,
+        }
+
+        impl TryFrom<BracketRace> for StandingsRace {
+            type Error = BracketError;
+
+            fn try_from(value: BracketRace) -> Result<Self, Self::Error> {
+                let player_1_result = value
+                    .player_1_result()
+                    .ok_or(BracketError::InvalidState)??;
+                let player_2_result = value
+                    .player_2_result()
+                    .ok_or(BracketError::InvalidState)??;
+                let outcome = value.outcome()?.ok_or(BracketError::InvalidState)?;
+                Ok(Self {
+                    player_1_id: value.player_1_id,
+                    player_2_id: value.player_2_id,
+                    player_1_result,
+                    player_2_result,
+                    outcome,
+                })
+            }
+        }
+
         for round in rounds {
             let round_races = round.races(conn)?;
-            if !all_races_complete(&round_races) {
-                // this either means we've reached the current round, or it means that
-                // we have a data issue, and in either case i'm giving up
+            if !is_rr && !round_races.iter().all(BracketRace::is_complete) {
```

Comment:
this is getting a little gross in trying to handle round robin and swiss the same tbh. 

i'm not that worried about it so far but it's a little annoying